src/content/components/UnassignedView/UnassignedView.tsx

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import { BugList } from "../BugList/BugList";
+import { useBugFetcher, BugQuery } from "../../hooks/useBugFetcher";
+import { Container } from "../ui/Container/Container";
+import { Loader, MiniLoader } from "../Loader/Loader";
+import { columnTransforms as cTrans } from "../BugList/columnTransforms";
+
+interface GetQueryOptions {
+  iteration: string;
+  metas: Array<{
+    id: string;
+    component: string;
+    priority?: string;
+    displayName?: string;
+  }>;
+  components: string[];
+}
+
+const COLUMNS = [
+  "id",
+  "type",
+  "summary",
+  "assigned_to",
+  "priority",
+  "cf_fx_iteration",
+  "last_change_time",
+];
+
+const getQuery = (options: GetQueryOptions): BugQuery => ({
+  include_fields: [
+    "id",
+    "summary",
+    "assigned_to",
+    "priority",
+    "status",
+    "keywords",
+    "type",
+    "component",
+    "cf_fx_iteration",
+    "last_change_time",
+  ],
+  rules: [
+    {
+      key: "cf_fx_iteration",
+      operator: "substring",
+      value: options.iteration,
+    },
+    {
+      key: "component",
+      operator: "anyexact",
+      value: options.components.join(","),
+    },
+    {
+      operator: "OR",
+      rules: [
+        {
+          key: "priority",
+          operator: "equals",
+          value: "P1",
+        },
+        {
+          key: "priority",
+          operator: "equals",
+          value: "P2",
+        },
+      ],
+    },
+    {
+      rules: [
+        {
+          key: "assigned_to",
+          operator: "equals",
+          value: "nobody@mozilla.org",
+        },
+      ],
+    },
+    { key: "keywords", operator: "notsubstring", value: "meta" },
+  ],
+});
+
+function sortBugs(bugs: any[]): any[] {
+  return bugs.sort((a, b) => {
+    let aIteration = cTrans.cf_fx_iteration(a.cf_fx_iteration);
+    let bIteration = cTrans.cf_fx_iteration(b.cf_fx_iteration);
+
+    let iterationComparison = 0;
+
+    if (aIteration != bIteration) {
+      if (aIteration === "--") {
```

Comment:
I'm pretty sure the null value for iteration is `---` not `--`. I think two dashes is for priority. Can you confirm this?

Reply:
I am using the `cf_fx_iteration` function in `src/content/components/BugList/columnTransforms.js` to parse iteration values, which returns `--` as a null value:
```
cf_fx_iteration(value) {
  return parseIteration(value) || "--";
},
```

Reply:
I see. That's just for UI. The parsing here is unnecessary, since this is only being done for sorting. Right now we're parsing every iteration string twice, once for sorting and again down the tree for rendering the text. But you can just compare the original iteration strings.
```js
parseFloat("110.1 - Jun 27 - 48") === 110.1;
parseFloat("---") === NaN;
parseFloat("") === NaN;
parseFloat() === NaN;
```

Also, bugs with no priority and no iteration are ending up at the top of the array. We have to check that they both actually have a priority before trying to compare them. This seems to work:
```js
[{cf_fx_iteration:NaN,priority:"P2",id:0},{cf_fx_iteration:"110.1",priority:"P2",id:1},{cf_fx_iteration:"110.1",priority:"P2",last_change_time:"2023-01-25T00:00:30.323Z",id:2},{cf_fx_iteration:"110.1",priority:"P2",last_change_time:"2023-01-25T00:00:31.423Z",id:3},{cf_fx_iteration:"110.1",priority:"P2",last_change_time:"2023-01-25T00:00:32.523Z",id:4},{cf_fx_iteration:"110.1",priority:"P2",last_change_time:"2023-01-25T00:00:33.931Z",id:5},{cf_fx_iteration:"110.2",priority:"P2",last_change_time:"2023-01-25T00:00:34.219Z",id:6},{cf_fx_iteration:NaN,priority:null,last_change_time:"2023-01-25T00:00:35.737Z",id:7},{cf_fx_iteration:"110.3",priority:"P1",last_change_time:"2023-01-25T00:00:36.152Z",id:8}]
.sort((a, b) => {
  let priorityComparison =
    a.priority && b.priority && a.priority.localeCompare(b.priority);
  if (priorityComparison) {
    return priorityComparison;
  }

  let iterationComparison = 0;
  if (a.cf_fx_iteration != b.cf_fx_iteration) {
    let aIteration = parseFloat(a.cf_fx_iteration);
    let bIteration = parseFloat(b.cf_fx_iteration);
    if (isNaN(aIteration)) {
      iterationComparison = Number(!isNaN(bIteration));
    } else if (isNaN(bIteration)) {
      iterationComparison = -1;
    } else {
      iterationComparison = aIteration - bIteration;
    }
  }

  return (
    iterationComparison ||
    Date.parse(b.last_change_time) - Date.parse(a.last_change_time)
  );
});
```